src/pages/NftDetailPage.tsx

```diff
@@ -14,25 +15,29 @@ const Container = styled.div``;
 export default () => {
   const { network, collection, tokenId } = useParams<Params>();
   const chainId = NETWORKS[network];
-  const { data, isLoading, isError } = useBrowseNftDetails(
+  const { nft, isLoading, isError } = useBrowseNftDetails(
     collection,
     tokenId,
     chainId
   );
 
+  const { metadata, isLoading: isMetadataLoading } = useMetadata(nft?.tokenUri);
```

Comment:
the hook returns all the usual react-query stuff, with `metadata` being `Metadata | undefined`